### DISCUSSION TITLE
Scale by `read`

### DISCUSSION THREAD

**Discussion by Naum-Tomov** [utwente-fmt/vercors#891] [Ideas]
We may want to scale a permission by read, but we may also want to not support it, because there exists an alternative for predicates, e.g. just using `Perm` or `Value`.

**Comment by bobismijnnaam** [comment]
Related: https://github.com/viperproject/silver/issues/534, https://github.com/viperproject/carbon/issues/376, https://github.com/utwente-fmt/vercors/issues/763

**Reply by pieter-bos** [reply]
Perhaps good to know in this context: we essentially normalize `Perm(_, read)` to `Value(_)` so that `read` itself does not occur as an expression -- rather the problem then is to define semantics for the result of scaling `Perm(_, p)` and `Value(_)` by `read`.
